[issue#109] clarify multiple enlistment of the same participant (#118)
eclipse/microprofile-lra: spec/src/main/asciidoc/microprofile-lra-spec.adoc

.Participant State Model
image::participant-state-model.png[align="center" link="participant-state-model"]

The participant can be enlisted only once per LRA instance. When participant
is about to be enlisted multiple times (e.g. client calls the same `@LRA` method

> several times) the follow-up enlistments of such participant MUST be ignored.

Reviewer: Request for clarification, same business logic is deployed to a cluster (like kubernetes) but different instances (e.g. pods) respond to same rest endpoint (e.g. /book) which was annotated with `@Lra`. I don't think it is feasible to prevent the second call from enlisting a new resource.

Reviewer 2: I believe it is because they will have the same identification which needs to be globally unique (e.g., in case of narayana the participant URIs/URLs will be the same for all pods as they are behind same load balancer or service in kubernetes)

Reviewer: Good point, I guess compensation URL of www.foo.com/<service_name>/compensation will attempt to be registered with a specifically identified LRA instance. Something for implementations to watch out for and not try to optimize for using verification on the client side alone if their implementation uses a client/server architecture.

Author: it's good point, I wasn't thinking this way when we were discussing this on the meeting. But still I agree with what Martin wrote here. The identification should be globally unique.

Reviewer 3: I agree with Martin, if the application cannot be coded to deal with multiple compensate/complete notifications then it is his responsibility to ensure the participant is registered with a unique id.

Reviewer: If the client side of the participant identification is somehow unique it is not cluster aware.

Also, it should not be getting multiple compensate notifications (for this particular requirements) as "follow-up enlistments of such participant MUST be ignored". Anyway this is really now just a note for implementations that they cannot expect to be able to process this requirement entirely on the client side (in client-server architectures).

Reviewer 3: Tom, I merged this PR but then realised it wasn't clear as to whether or not you were asking for changes?

Reviewer: No, it's fine thanks

> several times) the follow-up enlistments of such participant are to be ignored.

Reviewer 2: ```suggestion
several times) the follow-up enlistments of such participant MUST be ignored.
```
just to put an emphasis on what we expect?

Author: +1
